mixins/async-container/async-container-mixin.js

```diff
@@ -24,17 +24,14 @@ export const AsyncContainerMixin = superclass => class extends superclass {
 		super();
 		this._initializeAsyncState();
 		this.asyncPendingDelay = 0;
-		if (!this.asyncContainer) {
-			this.addEventListener('pending-state', this._handleAsyncItemState.bind(this));
+		this._handleAsyncItemState = this._handleAsyncItemState.bind(this);
+		if (!this.asyncContainerCustom) {
+			this.addEventListener('pending-state', this._handleAsyncItemState);
 		}
 	}
 
-	firstUpdated() {
-		if (!this.asyncContainer) return;
-		const container = this.asyncContainer();
```

Comment:
Only `d2l-dialog` and `d2l-dialog-fullscreen` were providing a `asyncContainer` method, so this is safe to remove.